shell/platform/darwin/ios/framework/Source/FlutterAppDelegate.mm

```diff
@@ -171,7 +188,14 @@ - (BOOL)application:(UIApplication*)application
   if ([_lifeCycleDelegate application:application openURL:url options:options]) {
     return YES;
   }
-  return [self openURL:url];
+  if (![self isFlutterDeepLinkingEnabled]) {
+    return NO;
+  }
+  [self openURL:url
+                options:options
+      completionHandler:^(BOOL success){
+      }];
+  return YES;
```

Comment:
can this method get the result from completion handler only?>>

this application:openURL:options: method is non-async,  i don't think its return value can be set from the completion handler so i used the deep link flag value.

Reply:
But should this method return before the completion handler is complete?  Before, this method didn't return until `pushRouteInformation` was actually sent, whereas now it would return instantly.
It seems like you'd want to spin the runloop until the completion handler was done, and then return the value from the completion handler.

Reply:
what's the best practice to "spin the runloop"?

I updated the code to wait for the completion handler but got failures : 

Failures for clang-tidy on /Volumes/Work/s/w/ir/cache/builder/src/flutter/shell/platform/darwin/ios/framework/Source/FlutterAppDelegate.mm:
/Volumes/Work/s/w/ir/cache/builder/src/flutter/shell/platform/darwin/ios/framework/Source/FlutterAppDelegate.mm:178:3: error: Waiting on a callback using a semaphore creates useless threads and is subject to priority inversion; consider using a synchronous API or changing the caller to be asynchronous [clang-analyzer-optin.performance.GCDAntipattern,-warnings-as-errors]
  178 |   dispatch_semaphore_wait(semaphore, 5 * NSEC_PER_SEC);